OOModel/src/expressions/MetaCallExpression.h

```diff
@@ -49,6 +52,22 @@ class OOMODEL_API MetaCallExpression: public Super<Expression>
 		 * Returns the meta definition used in this meta call if one exists.
 		 */
 		MetaDefinition* metaDefinition();
+
+		/**
+		 * Generates the tree resulting from this meta call and returns it.
+		 */
+		Declaration* generate();
```

Comment:
One public method sounds good. But don't call it generate. Call it `generatedTree` as it is just a getter. The fact that there is a cache, and the fact that the cache might be empty, thus triggering an on-the-fly generation, should be an implementation detail.